### DISCUSSION TITLE
Posting Code? Uploading Code?

### DISCUSSION THREAD

**Discussion by jaxepps** [community/community#28248] [New to GitHub]
Hi, I've never heard of GitHub before and am very inexperienced with this entire area. I made a project on Processing and it has a .pde file name. I'm looking to embed that on a website and various forums have led me to posting said project on GitHub in order to have running code as a viable link for people to click on my website. 

So what I think I'm trying to do is upload the existing code that I wrote in processing and somehow magically turn that into something clickable via GitHub. 

Any and all help is welcome, thanks!!

**Comment by radar-blip** [comment]
What your looking for probably is GitHub Pages. But first you must make a new repository, which will include your files. You may have to do a little editing to your .pde files.

Stay tuned for others to chime in. 👍🏼

**Reply by jaxepps** [reply]
This is totally the right direction thank you! I now have a repository and have uploaded my file... what's my next step?

**Reply by radar-blip** [reply]
Open your repository, open settings up at the top, and go from there. I think your repository must have a index.html in it, for everything to work correctly.

**Reply by jaxepps** [reply]
I'm not seeing anything in Settings to publish. How to add/ what is an index.html. Thanks again

**Reply by radar-blip** [reply]
Pick a branch source: 'home' or 'main' something like that, and a theme (if you need one).
But if you dont have an index.html you might need to create one. I'm hoping someone will chime in to help here, its been a while since i've done any html stuff.

**Reply by radar-blip** [reply]
You did it. 👍🏼

**Comment by jfullstackdev** [comment]
another one is Gist, which I'm using if I'm uploading content in LMS

this way, I just need to update the content on Gist and it will be reflected on all of the sites this gist is embedded
